### PR TITLE
OpenBSD Fixes/Cleanups

### PR DESCRIPTION
**Pull request by mrme0w**
OpenBSD has a rolling release cycle. A new release comes out every 6 months. Anything older than 1 year is no longer supported, nor does it receive patches. It's dead jim.

"OpenBSD only has support for 1 year. Removed old instructions and cleaned up instructions for 6.4 and 6.5."

### CODE REVIEW

**Comment by mrme0w** [comment]
OpenBSD's compilers already produce hardened code by default. Switch this off to avoid broken binaries.

Tested on OpenBSD 6.4/6.5/amd64

I'm used to CVS (I know.. I know...).. I'm still learning git so please excuse the two commits lumped into a single pull. :)

**Comment by moneromooo-monero** [comment]
Those options make the binary actually break ?

**Comment by mrme0w** [comment]
Yes, you end up with compiled binaries that won't run at all. They say the ELF format is wrong.

As mentioned before: adding hardening flags on OpenBSD is not nessasary, nor it is a good idea. Our compilers are patched to harden automatically. :)

**Comment by mrme0w** [comment]
Oh, are there plans for an OpenBSD buildbox at all? I noticed that there was some sort of effort back in January. I do have access to OpenBSD/amd64 (and OpenBSD/arm64 soon!) gear for building and testing.

**Comment by moneromooo-monero** [comment]
OK. Sounds like a compiler/linker bug if they don't run with those, but hey, sounds good ^_^
I'll ask about the openbsd machine.

File: README.md [hunk 1]
```diff
@@ -465,9 +465,11 @@ We expect to add Monero into the ports tree in the near future, which will aid i
 
 You will need to add a few packages to your system. `pkg_add cmake gmake zeromq cppzmq libiconv boost`.
 
+If you would like to run the test suite, you will also need to add the py-requests package to your system.
+
 The doxygen and graphviz packages are optional and require the xbase set.
 
-Build monero: `gmake release-static`
+Build monero: `env DEVELOPER_LIBUNBOUND_OLD=1 gmake release-static`
```
Comment (moneromooo-monero) [inline README.md]: That's obsolete. It's not used anymore. I will remove references to it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -657,38 +657,39 @@ else()
...
+  if (NOT OPENBSD) # OpenBSD's defaults are hardened by default
+    add_c_flag_if_supported(-mmitigate-rop C_SECURITY_FLAGS)
+    add_cxx_flag_if_supported(-mmitigate-rop CXX_SECURITY_FLAGS)
+    # linker
+    if (NOT WIN32)
+      # Windows binaries die on startup with PIE
+      add_linker_flag_if_supported(-pie LD_SECURITY_FLAGS)
+    endif()
+    add_linker_flag_if_supported(-Wl,-z,relro LD_SECURITY_FLAGS)
+    add_linker_flag_if_supported(-Wl,-z,now LD_SECURITY_FLAGS)
```
Comment (ston1th) [inline CMakeLists.txt]: OpenBSD does `Partial RELRO` by default, but not `Full RELRO`.

We could move the flag `-Wl,-z,now` out of the `if (NOT OPENBSD)` block to enable `Full RELRO`.

**Comment by moneromooo-monero** [comment]
> Yes, you end up with compiled binaries that won't run at all.

Are they running fine for you with the proposed change ?

**Comment by ston1th** [comment]
The changes by @mrme0w look good to me.
More info on the compiler behavior on OpenBSD can be found here: [clang-local(1)](https://man.openbsd.org/clang-local)

I compiled with my proposed change, so `Full RELRO` is active.
All tests pass and the compiled binaries seem to work fine again.

The only exception is `core_tests` which seems to take a lot of time (still running).

**Comment by dEBRUYNE-1** [comment]
@mrme0w - Can you rebase and squash the last two commits into one?

**Comment by moneromooo-monero** [comment]
ping, please squash as requested above.

**Comment by moneromooo-monero** [comment]
Superseded by https://github.com/monero-project/monero/pull/5773

+duplicate

**Pull request closed without merging**